system/security/plugins/jwtSecurity/jwtSecurity.cpp

```diff
@@ -0,0 +1,1129 @@
+                    }
+                    ISecResource* rs = rlist->addResource(rstr.str());
+                    SecAccessFlags requiredaccess = str2perm(required.str());
+                    rs->setRequiredAccessFlags(requiredaccess);
+                    rs->setDescription(description.str());
+                }
+                loc_iter->next();
+            }
+        }
+
+        return authmap;
+    }
+
+    virtual IAuthMap* createFeatureMap(IPropertyTree* authconfig, IEspSecureContext* secureContext = nullptr) override
+    {
+        CAuthMap* feature_authmap = new CAuthMap();
+
+        Owned<IPropertyTreeIterator> feature_iter;
+        feature_iter.setown(authconfig->getElements(".//Feature"));
+        ForEach(*feature_iter)
+        {
+            IPropertyTree* feature = nullptr;
+            feature = &feature_iter->query();
+            if (feature != nullptr)
+            {
+                StringBuffer pathstr, rstr, required, description;
+                feature->getProp("@path", pathstr);
+                feature->getProp("@resource", rstr);
+                feature->getProp("@required", required);
+                feature->getProp("@description", description);
+
+                ISecResourceList* rlist = feature_authmap->queryResourceList(pathstr.str());
+
+                if (rlist == nullptr)
+                {
+                    rlist = createResourceList(pathstr.str(), secureContext);
+                    feature_authmap->add(pathstr.str(), rlist);
+                }
+                if (!rstr.isEmpty())
+                {
+                    ISecResource* rs = rlist->addResource(rstr.str());
+                    SecAccessFlags requiredaccess = str2perm(required.str());
+                    rs->setRequiredAccessFlags(requiredaccess);
+                    rs->setDescription(description.str());
+                }
+            }
+        }
+
+        return feature_authmap;
+    }
+
+    virtual IAuthMap* createSettingMap(IPropertyTree* authConfig, IEspSecureContext* secureContext = nullptr) override
+    {
+        return nullptr;
+    }
+
+private:
+
+    /**
+     * Boolean indicating if one string is a prefix of another.  Case sensitive.
+     *
+     * @param   prefix      The prefix to test
+     * @param   s           The string to test against
+     *
+     * @return  true if prefix is a case-sensitive prefix of, or exactly matches, s.
+     */
+    static bool IsPrefixString(const std::string& prefix, const std::string& s)
+    {
+        auto res = std::mismatch(prefix.begin(), prefix.end(), s.begin());
+
+        return res.first == prefix.end();
+    }
+
+    /**
+     * Returns a copy of the string argument with particular whitespace
+     * characters removed from the end.
+     *
+     * @param   s           The string to process
+     *
+     * @return  New string with trailing whitespace characters removed.
+     */
+    static std::string TrimString(const std::string& s)
```

Comment:
Any reason these methods start with uppercase and others don't?  Methods are usually lowercase in most of our code.

Reply:
More copying and pasting madness (I confess to using that style in my own code).  Fixed.